packages/eventsub-base/src/EventSubBase.ts

```diff
@@ -390,6 +394,64 @@ export abstract class EventSubBase extends EventEmitter {
 		return await this._genericSubscribe(EventSubChannelUnbanSubscription, handler, this, userId);
 	}
 
+	/**
+	 * Subscribes to events that represent Shield Mode being activated in a channel.
+	 *
+	 * @param broadcaster The user for which to get notifications for when Shield Mode is activated in their channel.
+	 * @param moderator A user that has permission to read Shield Mode status in the broadcaster's channel.
+	 * @param handler The function that will be called for any new notifications.
+	 *
+	 * @beta
+	 */
+	async subscribeToChannelShieldModeBeginEvents(
+		broadcaster: UserIdResolvable,
+		moderator: UserIdResolvable,
+		handler: (event: EventSubChannelShieldModeBeginEvent) => void
+	): Promise<EventSubSubscription> {
+		const broadcasterId = this._extractUserIdWithNumericWarning(
+			broadcaster,
+			'subscribeToChannelShieldModeStartEvents'
+		);
+		const moderatorId = this._extractUserIdWithNumericWarning(moderator, 'subscribeToChannelShieldModeStartEvents');
+
+		return await this._genericSubscribe(
+			EventSubChannelShieldModeBeginSubscription,
+			handler,
+			this,
+			broadcasterId,
+			moderatorId
+		);
+	}
+
+	/**
+	 * Subscribes to events that represent Shield Mode being deactivated in a channel.
+	 *
+	 * @param broadcaster The user for which to get notifications for when Shield Mode is deactivated in their channel.
+	 * @param moderator A user that has permission to read Shield Mode status in the broadcaster's channel.
+	 * @param handler The function that will be called for any new notifications.
+	 *
+	 * @beta
+	 */
+	async subscribeToChannelShieldModeEndEvents(
+		broadcaster: UserIdResolvable,
+		moderator: UserIdResolvable,
+		handler: (event: EventSubChannelShieldModeEndEvent) => void
+	): Promise<EventSubSubscription> {
+		const broadcasterId = this._extractUserIdWithNumericWarning(
+			broadcaster,
+			'subscribeToChannelShieldModeEndEvents'
+		);
+		const moderatorId = this._extractUserIdWithNumericWarning(moderator, 'subscribeToChannelShieldModeEndEvents');
+
+		return await this._genericSubscribe(
+			EventSubChannelShieldModeEndSubscription,
+			handler,
+			this,
+			broadcasterId,
+			moderatorId
+		);
+	}
```

Comment:
These need to be added to the `EventSubListener` interface as well

Reply:
Done.